resources/js/components/declarations/data/DeclarationRow.ts

```diff
@@ -0,0 +1,49 @@
+import moment, { Moment } from 'moment';
+
+export interface IDeclarationRow {
+    amount: string;
+    description: string;
+    date: string;
+    gift: boolean;
+    file?: File;
+}
+
+export class DeclarationRow implements IDeclarationRow {
+    public static Empty(): DeclarationRow {
+        return new DeclarationRow;
+    }
+    public static ForFile(file: File): DeclarationRow {
+        const row = new DeclarationRow();
+        row.file = file;
+        return row;
+    }
+
+    public static FromDeclaration(row: DeclarationRow): DeclarationRow {
+        const newRow = new DeclarationRow();
+        newRow.file = row.file;
+        newRow.amount = row.amount;
+        newRow._date = row._date.clone();
+        newRow.description = row.description;
+        newRow.gift = row.gift;
+        return row;
```

Comment:
Ik weet niet of ik helemaal begrijp wat hier gebeurt, maar zou dit niet `return newRow` moeten zijn?

Reply:
Jup dat moet newRow zijn